test/EndToEndTests/Tests/Client/Build.Desktop/ContainmentTest/ContainmentTest.cs

```diff
@@ -782,7 +789,11 @@ public void QueryEntityAndExpandContainedEntity()
 
                             Assert.AreEqual(ODataReaderState.Completed, reader.State);
                             Assert.AreEqual(testCase.Value[0], entries.Count);
-                            Assert.AreEqual(testCase.Value[1], navigationLinks.Count);
+
+                            int expectedLinksCount = mimeType.Contains(MimeTypes.ODataParameterFullMetadata)
+                                ? testCase.Value[1]
+                                : testCase.Value[2];
+                            Assert.AreEqual(expectedLinksCount, navigationLinks.Count);

```

Comment:
Let's discuss.  I wouldn't expect expectedLinksCount to vary depending on the full versus minimal metadata.

Reply:
For the case of "Accounts(101)?$expand=MyPaymentInstruments($select=PaymentInstrumentID,FriendlyName)", full metadata response contains nav links for entities contained by PaymentInstruments, which are not supposed to be materialized in the minimal metadata response.

---
In reply to: [224148515](https://github.com/OData/odata.net/pull/1286#discussion_r224148515) [](ancestors = 224148515)

Reply:
:grinning: 

---
In reply to: [224632181](https://github.com/OData/odata.net/pull/1286#discussion_r224632181) [](ancestors = 224632181)